### PR TITLE
Doors Lock on RMB

### PR DESCRIPTION
**Pull request by Gyrandola**
## About The Pull Request

> Using keys on a door will now unlock on LMB, and lock on RMB.

image

It's not as evil as it sounds!!! Makes it quicker actually.

### CODE REVIEW

**Comment by Red-Honk** [comment]
Yes, that's a good idea, because it is a tad annoying trying to figure out if a door is locked or not, locks door kicks to open it it's locked, now that's annoying.

As well as: Is this locked?; Unlocks it someone rushes in, well shit.

**Comment by Cyprex** [comment]
I love this concept

File: code/game/objects/structures/mineral_doors.dm [hunk 1]
```diff
@@ -223,11 +223,7 @@
 				user.visible_message(span_warning("[user] smashes through [src]!"))
 			return
 		if(locked)
-			playsound(src, rattlesound, 100)
-			var/oldx = pixel_x
-			animate(src, pixel_x = oldx+1, time = 0.5)
-			animate(pixel_x = oldx-1, time = 0.5)
-			animate(pixel_x = oldx, time = 0.5)
+			door_rattle()
```
Comment (Cyprex) [inline code/game/objects/structures/mineral_doors.dm:226]: Mention that you're refactoring stuff in the PR description too

Comment (Cyprex) [reply code/game/objects/structures/mineral_doors.dm:226]: Also imo this could be named just rattle()- nitpicks I know!

**Comment by Hmeister-real** [comment]
you fool, you imbecile, this is already a thing!

Hidden deep in the zeth slopcode, if you sneak and left click with a door you can silently check if it's locked

File: code/game/objects/structures/mineral_doors.dm [hunk 2]
```diff
@@ -852,7 +864,8 @@
 	repair_skill = /datum/skill/craft/masonry
 
 /obj/structure/mineral_door/wood/donjon/stone/attack_right(mob/user)
-	return
+	if(user.get_active_held_item())
+		..()
```
Comment (Cyprex) [inline code/game/objects/structures/mineral_doors.dm:868]: Just remove this proc entirely, it's already handled by parent /obj/structure/mineral_door/wood/donjon/attack_right(mob/user)

File: code/game/objects/structures/mineral_doors.dm [hunk 3]
```diff
@@ -418,6 +426,17 @@
 					obj_integrity = max_integrity
 				user.visible_message(span_notice("[user] repaired [src]."), \
 				span_notice("I repaired [src]."))		
+/obj/structure/mineral_door/attack_right(mob/user)
+	user.changeNext_move(CLICK_CD_FAST)
```
Comment (Cyprex) [inline code/game/objects/structures/mineral_doors.dm:430]: By default there appears to be a 0.1s click cooldown- this changes it to 0.9(0.8 for fast+0.1 base). What's the purpose of this change?

File: code/game/objects/structures/mineral_doors.dm [hunk 4]
```diff
@@ -362,8 +358,20 @@
...
 
+/obj/structure/mineral_door/proc/door_rattle()
+	playsound(src, rattlesound, 100)
+	var/oldx = pixel_x
+	animate(src, pixel_x = oldx+1, time = 0.5)
+	animate(pixel_x = oldx-1, time = 0.5)
+	animate(pixel_x = oldx, time = 0.5)
+
 /obj/structure/mineral_door/attackby(obj/item/I, mob/user)
+	user.changeNext_move(CLICK_CD_FAST)
```
Comment (Cyprex) [inline code/game/objects/structures/mineral_doors.dm:369]: Same as below

**Comment by Cyprex** [comment]
Good to merge once the points are addressed
